## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 437210`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/17246519539


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>lctime</li>
    <li>lctime.dist</li>
    <li>python312Packages.liberty-parser</li>
    <li>python312Packages.liberty-parser.dist</li>
    <li>python313Packages.liberty-parser</li>
    <li>python313Packages.liberty-parser.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>lctime</li>
    <li>lctime.dist</li>
    <li>python312Packages.liberty-parser</li>
    <li>python312Packages.liberty-parser.dist</li>
    <li>python313Packages.liberty-parser</li>
    <li>python313Packages.liberty-parser.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>lctime</li>
    <li>lctime.dist</li>
    <li>python312Packages.liberty-parser</li>
    <li>python312Packages.liberty-parser.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.liberty-parser</li>
    <li>python313Packages.liberty-parser.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>lctime</summary>
<pre>  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/main.py", line 367 in pytest_runtestloop
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/main.py", line 343 in _main
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/main.py", line 289 in wrap_session
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/main.py", line 336 in pytest_cmdline_main
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/nix/store/ynfdybznd8w0h5kvqilmj0fv821mq4sq-python3.13-pluggy-1.6.0/lib/python3.13/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py", line 175 in main
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/nix/store/4li3hfhl4bk4005dmbfilvg4z9n84zzi-python3.13-pytest-8.4.1/lib/python3.13/site-packages/pytest/__main__.py", line 9 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg, scipy._lib._ccallback_c, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, charset_normalizer.md, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._mt19937, numpy.random._generator, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, scipy.special._specfun, scipy.special._comb, scipy.linalg._fblas, scipy.linalg._flapack, _cyutility, scipy._cyutility, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.optimize._direct, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, regex._regex, _cffi_backend, yaml._yaml, PIL._imaging, kiwisolver._cext, psutil._psutil_osx, psutil._psutil_posix, lz4._version, lz4.frame._frame, matplotlib.backends._macosx (total: 81)
/nix/store/p4bmhhd41hp01x5b4g8kv78c90lcpmf9-pytest-check-hook/nix-support/setup-hook: line 23:  5924 Segmentation fault: 11     /nix/store/gkspnzkgarw4qvbl4670p3jcbgg6psrg-python3-3.13.6/bin/python3.13 "${flagsArray[@]}"</pre>
</details>
<details>
<summary>python312Packages.liberty-parser</summary>
<pre>__________________________ test_parse_liberty_freepdk __________________________

    def test_parse_liberty_freepdk():
        import os.path
        lib_file = os.path.join(os.path.dirname(__file__), '../../test_data/gscl45nm.lib')
    
        data = open(lib_file).read()
    
        library = parse_liberty(data)
        assert isinstance(library, Group)
    
        library_str = str(library)
>       open('/tmp/lib.lib', 'w').write(library_str)
        ^^^^^^^^^^^^^^^^^^^^^^^^^
E       PermissionError: [Errno 13] Permission denied: '/tmp/lib.lib'

src/liberty/parser.py:460: PermissionError
=========================== short test summary info ============================
FAILED src/liberty/parser.py::test_parse_liberty_freepdk - PermissionError: [Errno 13] Permission denied: '/tmp/lib.lib'
========================= 1 failed, 45 passed in 2.54s =========================</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>lctime</li>
    <li>lctime.dist</li>
    <li>python313Packages.liberty-parser</li>
    <li>python313Packages.liberty-parser.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.liberty-parser</li>
    <li>python312Packages.liberty-parser.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.liberty-parser</summary>
<pre>__________________________ test_parse_liberty_freepdk __________________________

    def test_parse_liberty_freepdk():
        import os.path
        lib_file = os.path.join(os.path.dirname(__file__), '../../test_data/gscl45nm.lib')
    
        data = open(lib_file).read()
    
        library = parse_liberty(data)
        assert isinstance(library, Group)
    
        library_str = str(library)
>       open('/tmp/lib.lib', 'w').write(library_str)
        ^^^^^^^^^^^^^^^^^^^^^^^^^
E       PermissionError: [Errno 13] Permission denied: '/tmp/lib.lib'

src/liberty/parser.py:460: PermissionError
=========================== short test summary info ============================
FAILED src/liberty/parser.py::test_parse_liberty_freepdk - PermissionError: [Errno 13] Permission denied: '/tmp/lib.lib'
========================= 1 failed, 45 passed in 1.12s =========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/liberty-parser/default.nix
+++ b/pkgs/development/python-modules/liberty-parser/default.nix
@@ -1,0 +1,65 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitea,
+  setuptools,
+  lark,
+  numpy,
+  sympy,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "liberty-parser";
+  version = "0.0.25";
+  pyproject = true;
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "tok";
+    repo = "liberty-parser";
+    tag = version;
+    hash = "sha256-Nl+FRG93DeP1ctDphaTKZqkukEywmGprj6JORJQTunw=";
+  };
+
+  # Tests try to write to /tmp directly. use $TMPDIR instead.
+  postPatch = ''
+    substituteInPlace src/liberty/parser.py \
+      --replace-fail "/tmp" "$TMPDIR"
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    lark
+    numpy
+    sympy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  enabledTestPaths = [
+    "src/liberty/*.py"
+  ];
+
+  pythonImportsCheck = [
+    "liberty.parser"
+  ];
+
+  meta = {
+    description = "Liberty parser for Python";
+    homepage = "https://codeberg.org/tok/liberty-parser";
+    license = with lib.licenses; [
+      asl20
+      cc-by-sa-40
+      cc0
+      gpl3Plus
+    ];
+    maintainers = with lib.maintainers; [ eljamm ];
+    teams = with lib.teams; [ ngi ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -8123,6 +8123,8 @@ self: super: with self; {
 
   libear = callPackage ../development/python-modules/libear { };
 
+  liberty-parser = callPackage ../development/python-modules/liberty-parser { };
+
   libevdev = callPackage ../development/python-modules/libevdev { };
 
   libfdt = toPythonModule (

